### ISSUE TITLE
Profiling: IndexError occurs when using pypesto_problem read from hdf5-file

### ISSUE THREAD

**Issue by IvaEwert** [ICB-DCM/pyPESTO#483]
I tried to compute parameter profiles for my already executed parameter estimation provided as pypesto_problem and pypesto_result stored in a hdf5-file. 
When running my script (script and hdf5 file are attached) I get the following error message:
```
File "/usr/lib/python3.8/code.py", line 90, in runcode
    exec(code, self.locals)
  File "<input>", line 1, in <module>
  File "/snap/pycharm-community/214/plugins/python-ce/helpers/pydev/_pydev_bundle/pydev_umd.py", line 197, in runfile
    pydev_imports.execfile(filename, global_vars, local_vars)  # execute the script
  File "/snap/pycharm-community/214/plugins/python-ce/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/home/PycharmProjects/UNCOVer/models/Zhao_QuantBiol2020/profiling.py", line 24, in <module>
    result_profiling = profile.parameter_profile(
  File "/home/iva/pyPESTO/pypesto/profile/profile.py", line 107, in parameter_profile
    current_profile = walk_along_profile(
  File "/home/pyPESTO/pypesto/profile/profile.py", line 181, in walk_along_profile
    x_next = create_next_guess(x_now, i_par, par_direction,
  File "/home/pyPESTO/pypesto/profile/profile.py", line 76, in create_next_guess
    return next_guess(x, par_index, par_direction_, profile_options_,
  File "/home/pyPESTO/pypesto/profile/profile_next_guess.py", line 63, in next_guess
    return adaptive_step(x, par_index, par_direction, profile_options,
  File "/home/pyPESTO/pypesto/profile/profile_next_guess.py", line 161, in adaptive_step
    problem.fix_parameters(par_index, x[par_index])
  File "/home/pyPESTO/pypesto/problem.py", line 231, in fix_parameters
    self.normalize()
  File "/home/pyPESTO/pypesto/problem.py", line 178, in normalize
    if self.x_guesses_full.shape[1] != self.dim_full:
IndexError: tuple index out of range
```
The x_guesses_full array is empty and it seems that it is not included into the pypesto_problem during the hdf5 saving.
Fixed it with adding those lines to my script:
```
x_guesses_full = np.zeros((0, pypesto_problem.dim_full))
pypesto_problem.x_guesses_full = x_guesses_full
```
[issue.zip](https://github.com/ICB-DCM/pyPESTO/files/5376986/issue.zip)

**Comment by jvanhoefer** [comment]
Hey Iva, 
thank you for reporting the issue. Can you also provide a minimal example of _saving_ of this optimization runs, since the problem seems to be, that x_guesses are not stored in the first place.

**Comment by IvaEwert** [comment]
Hey, 
I added an example of the optimization and its saving as well as the corresponding PEtab files. I hope it is sufficient, if not, please contact me again.
[Example.zip](https://github.com/ICB-DCM/pyPESTO/files/5383662/Example.zip)

**Comment by paulstapor** [comment]
Not too long ago, the indexing changed when computing profiles:
Formerly, it was a binary indexing, e.g., [0, 0, 1, 1, ,1 ,0, ...]. indicating whether profiles should be computed or not.
Now, it's just a list of the parameter indices (e.g., [3,4,5, ...]) which should be regraded in profile computation.
Not sure, but might this be the reason?

**Comment by IvaEwert** [comment]
I think this was not the reason for this IndexError, but thanks anyway for indicating this change in the parameter indices! I used the rosenbrock.ipynb as a template and there binary indexing is still used I think.

**Comment by plakrisenko** [comment]
fixed in #489
